EIPS/eip-3540.md

```diff
@@ -93,88 +89,94 @@ The list of section headers is terminated with the *section headers terminator b
-| outputs          | 1 byte   | 0x00-0x7F    | number of stack elements the code section returns  |
-| max_stack_height | 2 bytes  | 0x0000-0x3FF | max height of operand stack during execution       |
-| code_section     | variable | n/a          | arbitrary bytecode                                 |
-| data_section     | variable | n/a          | arbitrary sequence of bytes                        |
+_note: `,` is a concatenation operator, `+` should be interpreted as "one or more" of the preceding item, and `*` should be interpreted as "zero or more" of the preceding item._
+
+#### Header
+
+| name                   | length   | value         | description                                                                             |
+|------------------------|----------|---------------|-----------------------------------------------------------------------------------------|
+| magic                  | 2 bytes  | 0xEF00        |                                                                                         |
+| version                | 1 byte   | 0x01          | EOF version                                                                             |
+| kind_types             | 1 byte   | 0x01          | kind marker for types size section                                                      |
+| types_size             | 2 bytes  | 0x0004-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the type section content      |
+| kind_code              | 1 byte   | 0x02          | kind marker for code size section                                                       |
+| num_code_sections      | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the number of the code sections             |
+| code_size              | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the code section content      |
+| kind_container         | 1 byte   | 0x03          | kind marker for container size section                                                  |
+| num_container_sections | 2 bytes  | 0x0001-0x00FF | 16-bit unsigned big-endian integer denoting the number of the container sections        |
+| container_size         | 2 bytes  | 0x0001-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the container section content |
+| kind_data              | 1 byte   | 0x04          | kind marker for data size section                                                       |
+| data_size              | 2 bytes  | 0x0000-0xFFFF | 16-bit unsigned big-endian integer denoting the length of the data section content (*)  |
+| terminator             | 1 byte   | 0x00          | marks the end of the header                                                             |
+
+(*) For not yet deployed containers this can be more than the actual content, see Data Section Lifecycle section in the contract creation EIP-???? (TBD + link).
```

Comment:
```suggestion
(*) For not yet deployed containers this can be greater than the actual content length, see Data Section Lifecycle section in the contract creation EIP-???? (TBD + link).
```